lib/client.js

```diff
@@ -159,11 +164,6 @@ var Client = function(options) {
   this._context = options.context || null;
   this.ready = false;
 
-  if (this._parent) {
-    this.ready = this._parent.ready;
-    return this; // shortcut handshake
-  }
-
   this.on('app.registered', function(data) {
```

Comment:
Using the instances API to fetch an instance, and then using the instances API from that instance?